src/common/http/http.cc

```diff
@@ -503,23 +503,13 @@ response_t HttpImpl::Post(
     response_t res(new beast::http::response<beast::http::string_body>);
     beast::http::async_read(stream, buffer, *res, yield);
 
-    // Gracefully close the socket.
-    // Receive an error code instead of throwing an exception if this fails, so
-    // we can ignore some expected not_connected errors.
-    boost::system::error_code ec;
-    stream.async_shutdown(yield[ec]);
-
-    if (ec) {
-      // not_connected happens sometimes so don't bother reporting it.
-      // stream_truncated also happens sometime and we choose to ignore the
-      // stream_truncated error, as recommended by the github thread:
-      // https://github.com/boostorg/beast/issues/824
-      if (ec != beast::errc::not_connected &&
-          ec != boost::asio::ssl::error::stream_truncated) {
-        spdlog::error("{}: HTTP error encountered: {}", __func__, ec.message());
-        return response_t();
-      }
-    }
+    spdlog::trace("{}: closing connection, response payload size {}", __func__,
+                  res->payload_size().value());
+    // Close the socket. We already got the response we need so close the socket
+    // directly. We choose not to use `ssl_stream::async_shutdown` since in some
+    // case, the HTTPS server does not participate with closing
+    // stream/connection. That would make the async_shutdown waiting forever.
+    beast::get_lowest_layer(stream).close();
```

Comment:
I got the cause of this change. This problem is rooted from asio's implementation and I think it is better to use timeout to wait ack of `close_notify` message. (But `async_shutdown` doesn't have this feature. ref https://github.com/chriskohlhoff/asio/issues/650)
My proposal is to note that this fix is like a workaround here and may be fixed in future work.

Reply:
I like the timeout idea.

Reply:
I saw that post as well. And my initial attempt, https://github.com/istio-ecosystem/authservice/pull/212/commits/eeb57760a099a06f98a0fb1e51472029e3f855da was trying to use a timer, but I didn't managed to get it work properly, lack of C++ and boost familiarity ... :( 

Added a tracker and a TODO. https://github.com/istio-ecosystem/authservice/issues/214

Reply:
I think that https://github.com/istio-ecosystem/authservice/commit/eeb57760a099a06f98a0fb1e51472029e3f855da won't work because the timer callback won't be executed after `async_shutdown` will have done ;) This is why we should wait the fix of boost/asio internal. Thanks.